vignettes/SEIRDV.Rmd

```diff
@@ -0,0 +1,280 @@
+depends on the fraction of the population that is susceptible, the fraction of
+the population that is infectious, and the odds of an infectious individual
+infecting a susceptible individual, $\beta$. This is represented in the
+first terms on the right-hand side of $\text{d}S/ \text{d}t$ and $\text{d}E/ \text{d}t$.
+Exposed individuals become infectious at rate $\kappa$. $1/ \kappa$ is
+the incubation time of the virus, i.e. how many days after exposure the individual
+becomes infectious to others. Infectious individuals can either recover or die
+because of the disease. Infectious individuals recover in $1/ \gamma$
+days, thus individuals recover at rate $\gamma$. Similarly, infectious
+individuals die due to the infection after $1/ \mu$ days, making the death rate $\mu$.
+Susceptible individuals are vaccinated at rate $\nu$. Among those who are immune
+(recovered and vaccinated), people lose their immunity against the virus at different
+rates depending on what type of immunity they acquired: $\delta_V$ for those vaccinated
+and $\delta_R$ respectively for those previously infected.
+
+In addition to these four main ODEs, the model also keeps track of the cumulative number 
+of cases ($C$) and the cumulative number of disease-related deaths ($D$) over time
+$$\frac{\text{d}C}{\text{d}t} = \beta S I,$$
+$$\frac{\text{d}D}{\text{d}t} = \mu I.$$
+
+The system is closed by specifying the initial conditions
+$$S(0) = S_0,\ E(0) = E_0,\ I(0) = I_0,\ R(0) = R_0,\ V(0) = V_0,\ C(0) = 0,\ D(0) = 0.$$
+In the implementation of the model in this package, the variables are normalized so 
+$S(t) + E(t) + I(t) + R(t) + V(t) + D(t) \equiv 1$ for any given $t$. This means that each of these state variables represents the fraction of the population in each of these states at a given point in time.
+
+We next illustrate how this model works using functionality available in `comomodels`.
+
+## The `SEIRDV` class
+
+We first create an `SEIRDV` object that we will use to run simulations.
+```{r}
+my_model <- SEIRDV()
+```
+
+Next, we set the parameter values for $\beta$, $\kappa$, $\gamma$, $\mu$, $\nu$,
+$\delta_V$ and $\delta_R$:
+```{r}
+params <- list(beta=1, kappa=0.9, gamma=0.5, mu=0.1, nu=0.4, delta_V=0.1,
+               delta_R=0.05)
+transmission_parameters(my_model) <- params
+```
+
+To check that the parameters have been set correctly, we call
+```{r}
+transmission_parameters(my_model)
+```
+
+We now need to set the initial conditions for the model. We only set the initial conditions $S(0)$, $E(0)$, $I(0)$, $R(0)$ and $V(0)$.
+```{r}
+initial_conditions(my_model) <- list(S0=0.99, E0=0, I0=0.01, R0=0, V0=0)
+```
+The initial conditions must sum to one. If they do not, an error is thrown.
+
+Now we simulate the system from $t=0$ days to $t=50$ days.
+```{r}
+intervention_parameters(my_model) <- list(starts=c(17, 35, 42),
+                                          stops=c(25, 39, 49),
+                                          coverages=c(1, 1, 1))
```

Comment:
I would add a subsection to the vignette explaining the intervention feature, how it can be used to supply step function vaccination rates, and why these are more realistic than a constant vaccination rate for all time. You could also add a plot of the intervention protocol resulting from these parameter choices.